test/__snapshots__/test.js.snap

```diff
@@ -1,5 +1,75 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`with processors AsyncCallbackInterface.webidl 1`] = `
+"\\"use strict\\";
+
+const conversions = require(\\"webidl-conversions\\");
+const utils = require(\\"./utils.js\\");
+
+const interfaceName = \\"AsyncCallbackInterface\\";
+
+exports.convert = function convert(value, { context = \\"The provided value\\" } = {}) {
+  if (!utils.isObject(value)) {
+    throw new TypeError(\`\${context} is not an object.\`);
+  }
+
+  const { wrapper } = {
+    wrapper(...args) {
+      let completion;
+      let thisArg = this;
+      let O = value;
+      let X = O;
+
+      try {
+        if (typeof O !== \\"function\\") {
+          X = O[\\"asyncMethod\\"];
+          if (typeof X !== \\"function\\") {
+            throw new TypeError(\`\${context} does not correctly implement AsyncCallbackInterface.\`);
+          }
+          thisArg = O;
+        }
+
+        for (let i = 0; i < args.length; i++) {
+          args[i] = utils.tryWrapperForImpl(args[i]);
+        }
+        if (args.length < 0) {
+          for (let i = args.length; i < 0; i++) {
+            args[i] = undefined;
+          }
+        } else if (args.length > 0) {
+          args.length = 0;
+        }
+
+        let callResult = Reflect.apply(X, thisArg, args);
+
+        callResult = Promise.resolve(callResult).then(
+          value => {
+            value = conversions[\\"any\\"](value, { context: context + \\" promise value\\" });
+
+            return value;
+          },
+          reason => reason
+        );
+
+        return callResult;
+      } catch (err) {
+        return Promise.reject(err);
+      }
+    }
+  };
+
+  Object.defineProperties(wrapper, {
+    [utils.wrapperSymbol]: { value },
```

Comment:
Why do we need a wrapper symbol?

Reply:
This is to ensure that the wrapper function doesn’t leak to user code.

Reply:
> This is to ensure that the wrapper function doesn’t leak to user code.

Can you explain how that would happen if the wrapper symbol was omitted?

Reply:
Well, if there’s a function like:
```webidl
partial interface EventTarget {
	sequence<EventListener> getEventListeners(DOMString type);
};
```

Then the wrapper symbol handles the conversion back to the user provided object is handled by **WebIDL2JS** instead of by **JSDOM**.

There’s also `NodeIterator` and `TreeWalker` that expose a `NodeFilter` attribute.